gluecodium/src/main/java/com/here/gluecodium/validator/LimeOverloadsValidator.kt

```diff
@@ -41,10 +43,29 @@ internal class LimeOverloadsValidator(
                 logger.error(limeFunction, "function has conflicting overloads")
                 false
             }
-            validateConstructors && signatureResolver.hasConstructorSignatureClash(limeFunction) -> {
+            validateCustomConstructors && signatureResolver.hasConstructorSignatureClash(limeFunction) -> {
                 logger.error(limeFunction, "constructor has conflicting overloads")
                 false
             }
             else -> true
         }
+
+    private fun validateFieldConstructor(
+        limeFieldConstructor: LimeFieldConstructor,
+        signatureResolver: LimeSignatureResolver
+    ): Boolean {
+        if (validateCustomConstructors) {
+            if (signatureResolver.hasConstructorSignatureClash(limeFieldConstructor.asFunction())) {
+                logger.error(limeFieldConstructor, "field constructor has conflicting overloads")
+                return false
+            }
+        } else {
```

Comment:
Maybe we still should validate non custom constructors?
Or does heaving custom ones means there is no defaults?

Reply:
If either "field" or "custom" constructors are present, there is no implicitly generated constructors anymore. So no signature/name clash with those would be possible.